### DISCUSSION TITLE
Search at Tunein music provider

### DISCUSSION THREAD

**Discussion by cociweb** [music-assistant/support#2639] [Feature requests and ideas]
Hello,

I've checked on several place but I didn't find any information about the search function/ ha search service for my problem
Unfortunatelly, the tunein provider does not return any findings for the radios neither by the GUI nor by the HA service call. 
(As I see it is not implemented in the tunein provider). - Maybe the `Search.ashx?query=` endpoint can be used for this purpose.

The question is that I don't know how should we handle it: as a FR or a bug? (is it intended to left it out or not?)

Thank's in advance.

**Comment by OzGav** [comment]
This is expected behaviour. Tunein only shows the favourites. If you want to search use the radio browser provider.

**Reply by cociweb** [reply]
hmm somehow it does not return those findings what I've expected on the search endpoint of radiotime

**Reply by OzGav** [reply]
Sorry I don’t understand this. What is radio time? What are you trying to do now? Are we still talking about TuneIn?

**Reply by cociweb** [reply]
well, tunein (frontend, brand) uses radiotime API. (which also belongs their corporate somehow)...

[Their  api](https://github.com/music-assistant/server/blob/stable/music_assistant/providers/tunein/__init__.py#L279) has several endpoints: browse, tune,... and search. All of them is used by the tunein provider except the search ( `search.ashx?q=` )

[Here](https://github.com/core-hacked/tunein-api/blob/master/endpoints/search-endpoint/README.md) you can find the unofficial documentation of it.

**Reply by OzGav** [reply]
That is not supported as I said. You can change this to a feature request.

**Comment by cociweb** [comment]
Related PRs: https://github.com/music-assistant/server/pull/2204 and https://github.com/music-assistant/beta.music-assistant.io/pull/31

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
